lib/ibmdb.js

```diff
@@ -536,116 +536,206 @@ IBMDB.prototype.updateOrCreate = IBMDB.prototype.save =
+        return callback && callback(err);
+      }
+      executeWithConnection(conn, function(err, data, meta) {
+        if (err) {
+          conn.rollbackTransaction(function(err) {
+            conn.close(function() {});
+            return callback && callback(err);
+          });
+        } else {
+          options.transaction = undefined;
+          conn.commitTransaction(function(err) {
+            if (err) {
               return callback && callback(err);
-            });
-          } else {
-            options.transaction = undefined;
-            conn.commitTransaction(function(err) {
-              if (err) {
-                return callback && callback(err);
-              }
+            }
 
-              conn.close(function() {});
-              return callback && callback(null, data, meta);
-            });
-          }
-        });
+            conn.close(function() {});
+            return callback && callback(null, data, meta);
+          });
+        }
       });
-    }
-  };
+    });
+  }
+};
 
 /**
- * Create the table for the given model
+ * Replace if the model instance exists with the same id
+ * or create a new instance
  *
  * @param {string} model The model name
- * @param {Object} [options] options
- * @param {Function} [cb] The callback function
+ * @param {Object} data The model instance data
+ * @param {Object} options for this function call
+ * @param {Function} [callback] The callback function
  */
-IBMDB.prototype.createTable = function(model, options, cb) {
-  debug('IBMDB.prototype.createTable ', model, options);
-  cb();
-};
+IBMDB.prototype.replaceOrCreate = function(model, data, options, callback) {
+  debug('IBMDB.prototype.replaceOrCreate (enter): model=%j, data=%j, ' +
+        'options=%j\n', model, data, options);
+  var self = this;
+  var idName = self.idName(model);
+  var stmt;
+  var tableName = self.tableEscaped(model);
+  var meta = {};
+
+  function executeWithConnection(connection, cb) {
+    // Execution for replaceOrCreate requires running 3
+    // separate SQL statements. The last depends on the
+    // result of the first couple.
+    var where = {};
+    where[idName] = data[idName];
+
+    var countStmt = new ParameterizedSQL('SELECT COUNT(*) AS CNT FROM ');
+    countStmt.merge(tableName);
+    countStmt.merge(self.buildWhere(model, where));
+    countStmt.noResults = false;
+
+    connection.query(countStmt, function(err, countData) {
+      debug('IBMDB.prototype.replaceOrCreate stmt: %j data: %j err: %j\n',
+            countStmt, countData, err);
+      if (err) return cb(err);
+
+      if (countData[0]['CNT'] > 0) {
+        // remove existing to replace with a new insert
+        stmt = self.buildDelete(model, where);
+        stmt.noResults = true;
+        connection.query(stmt, function(err, res) {
+          debug('IBMDB.prototype.replaceOrCreate stmt: %j data: %j err=%j\n',
+                stmt, res, err);
 
-IBMDB.prototype.buildIndex = function(model, property) {
```

Comment:
Nevermind. Ignore the comment. Looked at the source code instead of the file changes and everything is fine. Reshuffling of these functions made it look like it was removed.